### ISSUE TITLE
[Feature]: Support snapshot 

### ISSUE THREAD

**Issue by xiaofan-luan** [milvus-io/milvus#37626]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

We already support backup on milvus 
https://github.com/zilliztech/milvus-backup

The backup copies all the data to another S3 bucket and could be slower for some use cases.
If we can create a snapshot in less than 10s that could be great as a way to store historical data. 

### Describe the solution you'd like.

Snapshot API:
create snapshot -> leave one snapshot version 
list snapshot -> list all the snapshot version
describe snapshot -> given snapshot version, given it's description and time created
drop snapshot -> clean unused snapshot
restore snapshot  ->  revert the collection to an old snapshot

When we call create snapshot, here is what happened
1. flush the collection
2. list all the files, stored it into snapshot meta

When doing garbage collection, all the file in snapshot meta can not be expired. 

When restore happend, we actually create a new snapshot and update the file list with the restored version.

**Comment by xiaofan-luan** [comment]
This is actually to have. Any one want to take it?

**Comment by nileshsironja** [comment]
Do the database locking required after even flushing all the collections??

**Comment by xiaofan-luan** [comment]
all the snapshot applied on meta so I guess there is no need to lock the collections

**Comment by nileshsironja** [comment]
Okay, I'm looking into application consistent snapshots with Milvus database.
That means whatever the data an application writes to the storage, must be part of consistent snapshot.
So even when I do flushSync(), a database lock has to be taken for not serving any further write operation to make the snapshot consistent. Right?

**Comment by nileshsironja** [comment]
Please let me know if any kind of lock available in Milvus, to lock the database.

**Comment by nileshsironja** [comment]
Pause Write Operations Temporarily:

You can use Milvus's administrative commands or APIs to pause write operations. For example, you might use a command like:
milvusctl pause-writes

Or, if Milvus provides an API endpoint for this, you can send an HTTP request:
curl -X POST http://<milvus-server>/admin/pause-writes

**Comment by xiaofan-luan** [comment]
you can set the write quota to 0 if you do need to disable write. 
chekc the config here https://milvus.io/docs/configure_quotaandlimits.md#quotaAndLimits-related-Configurations

**Comment by nileshsironja** [comment]
Yeah, write quota configuration can easily done with birdwatcher or etcd or python client. 
Thanks.

**Comment by nileshsironja** [comment]
Added in values.yaml

----
milvus:
  config:
    quotaAndLimits:
      limitWriting:
        forceDeny: true
----

And did upgrade:
helm upgrade my-release milvus/milvus -f values.yaml

Still I see milvus writes are happening.

**Comment by nileshsironja** [comment]
Is there any API call which can I do to avoid writing to database?

**Comment by xiaofan-luan** [comment]
not yet. We use quota limit change to avoid write.

**Comment by xiaofan-luan** [comment]
but feel free to add one

**Comment by nileshsironja** [comment]
Can you please point me out the exact configuration which deny the writes? 

So far I had tried quotaandlimits.limitwriting.forcedeny=true and quotaandlimits.enabled=false.
But these still allow the writes.

Thanks in advance!

**Comment by nileshsironja** [comment]
Just an update. Configuration quotaandlimits.limitwriting.forcedeny=true is working.

**Comment by kaaancan** [comment]
For anybody who is searching for a official tool, https://github.com/zilliztech/milvus-backup
